core/src/main/scala/scalaz/NonEmptyList.scala

```diff
@@ -177,6 +177,17 @@ sealed abstract class NonEmptyListInstances extends NonEmptyListInstances0 {
       override def foldMap[A, B](fa: NonEmptyList[A])(f: A => B)(implicit M: Monoid[B]) =
         Foldable[IList].foldMap(fa.list)(f)(M)
 
+      override def traverse1[F[_], A, B](fa: NonEmptyList[A])(f: A => F[B])(implicit F: Apply[F]) = {
+        val revOpt: Maybe[F[NonEmptyList[B]]] =
+          F.unfoldrOpt[IList[A], B, NonEmptyList[B]](fa.list)(_ match {
+            case ICons(a, as) => Maybe.just((f(a), as))
+            case INil() => Maybe.empty
+          })(Reducer.ReverseNonEmptyListReducer[B])
+
+        val rev: F[NonEmptyList[B]] = revOpt getOrElse sys.error("Head cannot be empty")
```

Comment:
This again uses an unsafe operation (`sys.error`).

What about
```scala
F.apply2(f(fa.head), fa.tail.traverse)(NonEmptyList.nel)
```
? (Which actually does reuse `IList.traverse`.) Should remain short-circuiting if `apply2` is, and only adds constant stack, so remains stack safe.

Reply:
That makes sense to me. Since`fa.tail.traverse` expects `F` to be an `Applicative` instead of an `Apply`, is there a way we can convert the `Apply` passed into `traverse1`?